core/src/main/java/org/jboss/modcluster/mcmp/impl/DefaultMCMPHandler.java

```diff
@@ -800,6 +802,10 @@ void setEstablished(boolean established) {
         private synchronized Socket getConnection() throws IOException {
             if (this.socket == null || this.socket.isClosed()) {
                 this.socket = this.socketFactory.createSocket();
+                // refresh ip for the proxy?
+                if (REFRESH_PROXY_IP && this.getState() != State.OK ) {
+                    this.socketAddress = new InetSocketAddress(this.socketAddress.getHostName(), this.socketAddress.getPort());
```

Comment:
We don't want to do this if the address in question was specified as an IP address - as this is both unnecessary and would force an undesirable reverse name lookup.

Reply:
Additionally, if a security manager is enabled, resolved addresses are cached forever by default.  If this is the case, creating a new InetAddress will resolve the address to the same value.

Reply:
JVM level DNS cache settings will impact behavior here of course, but the problem is current modcluster code forces permanent IP caching for the proxy regardless of the JVM setting.

In most environments, the proxy host resolution does not change so I don't know that default behavior needs to be changed so I put this disabled by default behind a system property.  It should only be enabled if repeat host resolution is desired so I don't think it would be enabled if an IP address is configured instead for a reverse lookup to occur in that case.

Reply:
https://docs.oracle.com/javase/7/docs/api/java/net/InetAddress.html look for InetAddress Caching
(Just a note on the topic).

Reply:
Adding java.security.Security.setProperty("networkaddress.cache.ttl" , "0");
Seems to disable the cache but it applies to whole JVM do we want do that?